### PR TITLE
[SDK-2684] Run tests against Node-OIDC in Browserstack (manually)

### PR DESCRIPTION
**Pull request by frederikprijck**
This PR updates our e2e tests to run against node-oidc-provider as well as prepare then to be executable on Browserstack.

Apparantly, Firefox has issues with secure context when running on Browserstack using `http://localhost`, but not when using `http://127.0.0.1`.
It looks like firefox detects that, even tho the files are served through localhost, those files are not originating from the same machine as where the browser is running. To solve this, we are using `127.0.0.1` instead of `localhost` in our tests now.

### CODE REVIEW

File: cypress/integration/getTokenSilently.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ describe('getTokenSilently', () => {
     }).as('tokenApiCheck');
 
     whenReady();
-    cy.toggleSwitch('use-form-data');
+
```
Comment (frederikprijck) [inline cypress/integration/getTokenSilently.js:23]: As we use the node-oidc provider, it only supports form data, which is enabled in `resetTests` now.

File: static/index.html [hunk 2]
```diff
@@ -512,6 +530,8 @@ <h3 class="mb-5">Other switches</h3>
                 useFormData: this.useFormData
               })
             );
+
+            this.initializeClient();
```
Comment (frederikprijck) [inline static/index.html:534]: Whenever the form is saved, we now reinitialice the client.

File: static/index.html [hunk 3]
```diff
@@ -402,8 +417,11 @@ <h3 class="mb-5">Other switches</h3>
                 access_tokens: []
               },
               {
-                audience: 'https://' + data.domain + '/api/v2/',
-                scope: 'read:rules',
+                audience:
+                  data.domain?.indexOf('http') > -1
+                    ? data.domain + '/api/v2/'
+                    : 'https://' + data.domain + '/api/v2/',
```
Comment (frederikprijck) [inline static/index.html:423]: Ensure we also support setting the domain with `http(s)` explicitly

File: static/index.html [hunk 4]
```diff
@@ -27,6 +27,19 @@ <h1 class="mb-5">Auth0 SPA JS Playground</h1>
...
+            </button>
+
+            <button
+              class="btn btn-info"
+              @click="useNodeOidcProvider"
+              data-cy="use-node-oidc-provider"
+            >
+              Use Node OIDC Provider
+            </button>
+          </div>
```
Comment (frederikprijck) [inline static/index.html:42]: This allows toggling between Auth0 and Node OIDC. The latter is mostly for the automated tests whereas Auth0 will be used when manually interacting with the playground.

File: scripts/oidc-provider.js [hunk 5]
```diff
@@ -1,10 +1,26 @@
...
+    { name: 'noop', requestable: false },
+    new Check('foo', 'bar', ctx => {
+      if (ctx.query?.scope?.includes('offline_access')) {
+        ctx.oidc.params.scope = `${ctx.oidc.params.scope} offline_access`;
+      }
+      return Check.NO_NEED_TO_PROMPT;
+    })
+  ),
+  0
+);
```
Comment (frederikprijck) [inline scripts/oidc-provider.js:17]: By default, Node-OIDC does not issue a refresh token unless you set `prompt=consent`. This is a bit different as to how Auth0 works and was causing some troubles with some tests.

This change now ensures refresh tokens are issues whenever `offline_access` is requested as a scope.

File: cypress/support/commands.js [hunk 6]
```diff
@@ -75,8 +87,21 @@ Cypress.Commands.add('loginNoCallback', () => {
...
+Cypress.Commands.add('fixCookies', () => {
+  // Temporary fix for https://github.com/cypress-io/cypress/issues/6375
+  if (Cypress.isBrowser('firefox')) {
+    cy.getCookies({ log: false }).then(cookies =>
+      cookies.forEach(cookie => cy.clearCookie(cookie.name, { log: false }))
+    );
+    cy.log('clearCookies');
+  } else {
+    cy.clearCookies();
+  }
```
Comment (frederikprijck) [inline cypress/support/commands.js:106]: Our tests where failing on firefox, which appears to be a known thing in Cypress that isnt solved yet.

**Review by stevehobbsdev — DISMISSED** [review]
🎉

**Review by stevehobbsdev — APPROVED** [review]

**Pull request merged**